sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SyncPollResponse.java

```diff
@@ -0,0 +1,116 @@
+final class SyncPollResponse<T, U> {
+    // AsyncPollResponse is a commonly used class, use a static logger.
+    private static final ClientLogger LOGGER = new ClientLogger(SyncPollResponse.class);
+    private final PollingContext<T> pollingContext;
+    private final BiFunction<PollingContext<T>, PollResponse<T>, T> cancellationOperation;
+    private final Function<PollingContext<T>, U> fetchResultOperation;
+    private final PollResponse<T> pollResponse;
+
+    /**
+     * Creates SyncPollResponse.
+     *
+     * @param pollingContext the polling context
+     * @param cancellationOperation the cancellation operation if supported by the service
+     * @param fetchResultOperation the operation to fetch final result of long-running operation, if supported
+     *                             by the service
+     */
+    SyncPollResponse(PollingContext<T> pollingContext,
+                     BiFunction<PollingContext<T>, PollResponse<T>, T> cancellationOperation,
+                     Function<PollingContext<T>, U> fetchResultOperation) {
+        this.pollingContext = Objects.requireNonNull(pollingContext,
+                "'pollingContext' cannot be null.");
+        this.cancellationOperation = Objects.requireNonNull(cancellationOperation,
+                "'cancellationOperation' cannot be null.");
+        this.fetchResultOperation = Objects.requireNonNull(fetchResultOperation,
+                "'fetchResultOperation' cannot be null.");
+        this.pollResponse = this.pollingContext.getLatestResponse();
+    }
+
+    /**
+     * Represents the status of the long-running operation at the time the last polling operation finished successfully.
+     * @return A {@link LongRunningOperationStatus} representing the result of the poll operation.
+     */
+    public LongRunningOperationStatus getStatus() {
+        return pollResponse.getStatus();
+    }
+
+    /**
+     * The value returned as a result of the last successful poll operation. This can be any custom user defined object,
+     * or null if no value was returned from the service.
+     *
+     * @return T result of poll operation.
+     */
+    public T getValue() {
+        return pollResponse.getValue();
+    }
+
+    /**
+     * Cancels the remote long-running operation if cancellation is supported by the service.
+     *
+     * @return T result of cancel operation.
+     */
+    public T cancelOperation() {
+        try {
+            return this.cancellationOperation
+                    .apply(this.pollingContext, this.pollingContext.getActivationResponse());
+        } catch (RuntimeException ex) {
+            throw LOGGER.logExceptionAsError(Exceptions.propagate(ex));
+        }
+    }
+
+    /**
+     * Fetches the final result of long-running operation if it
+     * is supported by the service. If the long-running operation is not completed, null will be returned.
+     *
+     * @return U result of fetch result operation.
+     */
+    public U getFinalResult() {
```

Comment:
Why does the poll response type contain the final result? It should only have the poll response. So, not sure we need this type since we have `PollResponse<T>` that doesn't have any async APIs and can be used by both sync and async pollers.